### PR TITLE
Macaroon endpoint

### PR DESCRIPTION
**Pull request by jrwren**

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/690/
Test FAILed.

**Comment by mhilton** [comment]
:+1: LGTM, once the  dependencies are fixed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/691/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/692/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/693/
Test PASSed.

File: internal/v4/api.go [hunk 1]
```diff
@@ -633,3 +635,13 @@ func (h *Handler) serveChangesPublished(_ http.Header, r *http.Request) (interfa
 	}
 	return results, nil
 }
+
+var IdentityLocation string = "identity"
```
Comment (mhilton) [inline internal/v4/api.go]: I'm pretty sure we no longer need this.

**Comment by mhilton** [comment]
:+1: LGTM (modulo the unused variable)

**Comment by urosj** [comment]
:+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/694/
Test PASSed.

**Comment by jrwren** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**